src/WebJobs.Script/Host/Kubernetes/KubernetesClient.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class KubernetesClient
+    {
+        private const string HttpLeaderEndpointKey = "HTTP_LEADER_ENDPOINT";
+        private readonly HttpClient _httpClient;
+        private readonly string _httpLeaderEndpoint;
+
+        public KubernetesClient(IEnvironment environment, HttpClient httpClient = null)
+        {
+            _httpClient = httpClient ?? new HttpClient();
+            _httpLeaderEndpoint = environment.GetEnvironmentVariable(HttpLeaderEndpointKey);
+        }
+
+        public async Task<KubernetesLockHandle> GetLock(string lockName)
+        {
+            var request = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Get,
+                RequestUri = GetRequestUri($"?name={lockName}")
+            };
+
+            var response = await _httpClient.SendAsync(request);
+
+            response.EnsureSuccessStatusCode();
+
+            var responseString = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<KubernetesLockHandle>(responseString);
+        }
+
+        public async Task<KubernetesLockHandle> TryAcquireLock (string lockId, string ownerId, TimeSpan lockPeriod, CancellationToken cancellationToken)
+        {
+            var lockHandle = new KubernetesLockHandle();
+            var request = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = GetRequestUri($"/acquire?name={lockId}&owner={ownerId}&duration={lockPeriod.TotalSeconds}&renewDeadline=10"),
```

Comment:
take `renewDeadline` as input as well?

Reply:
Sure. From where should I take it?

Reply:
I see, this not exposed in `IDistributedLockManager` . Move `renewDeadline` to a constant.